Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h

```diff
@@ -0,0 +1,56 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/** @file Digit.h
+ * C++ simple Muon MCH digit.
+ * @author  Michael Winn
+ */
+
+#ifndef ALICEO2_MCH_DIGIT_H_
+#define ALICEO2_MCH_DIGIT_H_
+
+#include "CommonDataFormat/TimeStamp.h"
+
+namespace o2
+{
+namespace mch
+{
+
+// \class Digit
+/// \brief MCH digit implementation
+using DigitBase = o2::dataformats::TimeStamp<double>;
+class Digit : public DigitBase
+{
+ public:
+  Digit() = default;
+
+  Digit(int pad, double adc); //check if need uint32_to
+  ~Digit() = default;
+
+  int getPadID() const { return mPadID; }
+  void setPadID(int pad) { mPadID = pad; }
+
+  double getADC() const { return mADC; }
+  void setADC(double adc) { mADC = adc; }
+
+  double getTimeStamp() { return mTime; }
+  void setTimeStamp(double time) { mTime = time; }
+
+ private:
+  int mPadID;
+  double mADC;
+  double mTime;
```

Comment:
I know you've not started the work on the timing aspect of the problem, but why is there both a mTime data member and a derivation from a TimeStamp ?

Reply:
Yes, I removed the mTime for the time being. Presumably, for MCH, it is not needed.